### PR TITLE
Add C support

### PR DESCRIPTION
**Pull request by pranavrao145**
Apologies for the swarm of PRs, it's just easier to add/fix stuff now that I'm a bit more familiar with the codebase :)

This PR basically takes #166 and replicates it for C (minus the tests for the class function, as that doesn't exist).

### CODE REVIEW

**Comment by ThePrimeagen** [comment]
I'm still out for Christmas and the new year. I won't be doing much reviewing of anything until after then.

File: queries/c/refactoring.scm
Comment (teddylear) [inline queries/c/refactoring.scm:18]: @pranavrao145 Ahhh sorry I should have reviewed this before I just merged the inline node PR. I moved statements to inline Nodes in Treesitter (see typescript or golang treesitter files). I'll hold off on merging moving everything to Inline nodes until this is updated and merged. Lmk if you want me to update PR as I broke it.

**Review by teddylear — CHANGES_REQUESTED** [review]
@pranavrao145 See comment above.

**Comment by pranavrao145** [comment]
No worries, I will try to make the fixes myself and I'll let you know if I need any help. Thanks!

File: queries/c/refactoring.scm
Comment (pranavrao145) [reply queries/c/refactoring.scm]: I've gone ahead and made some changes to inline the nodes for C. Luckily, C uses the same TS objects as CPP so there wasn't a lot of work required. :)

**Pull request merged**
